src/main/play-25/uk/gov/hmrc/cache/TimeToLiveProvider.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cache
+
+import java.util.concurrent.TimeUnit
+
+import javax.inject.{Inject, Provider}
+import play.api.{Configuration, Logger}
+
+import scala.concurrent.duration.{Duration, MINUTES}
+
+class TimeToLiveProvider @Inject()(configuration: Configuration) extends Provider[TimeToLive]{
+  override val get: TimeToLive = {
+    val oldConfiguration = configuration.getLong("cache.expiryInMinutes")
+      .map( value => Duration(value, MINUTES) )
+    oldConfiguration.foreach { _ =>
+      Logger.warn(
+        """Application use `cache.expiryInMinutes` deprecated in mongo-caching 6.x -
+          |please migrate configuration according to README""".stripMargin
+      )
+    }
+    val newKey = "cache.expiry"
+    val newConfiguration = Duration(configuration.getMilliseconds(newKey).get, TimeUnit.MILLISECONDS)
```

Comment:
Naked `get` on `Option` here. It throws when `cache.expiry` is not provided in configuration!
```suggestion
    val newConfiguration = configuration.getMilliseconds(newKey).fold(5.minutes){ expiry => Duration(expiry, TimeUnit.MILLISECONDS)
```

Reply:
There is default value in `referece.conf` with 5 minutes. There would be serous problems with configuration if this get fail.

Reply:
This change of property name is a bit error-prone. 
Developers might easily overlook the warning. Especially if they forget to change this setting in app-config - you need to explicitly search kibana to see it. 
Deployment and tests would succeed and the value will fall back to 5 minutes.
I would